src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -530,40 +535,44 @@ async function ProcessHTTPS(frb: FunctionsRuntimeBundle, trigger: EmulatedTrigge
 
 async function ProcessBackground(
   frb: FunctionsRuntimeBundle,
-  stubbedAdminModule: typeof admin,
   trigger: EmulatedTrigger
 ): Promise<void> {
   const { Change } = require("firebase-functions");
   new EmulatorLog("SYSTEM", "runtime-status", "ready").log();
 
+  if (!app) {
+    new EmulatorLog("SYSTEM", "admin-not-initialized", "").log();
+    return;
+  }
+
   const proto = frb.proto;
 
-  stubbedAdminModule.firestore().settings({});
-  const snapshot_ = (stubbedAdminModule.firestore() as any).snapshot_;
+  adminModuleProxy.firestore().settings({});
+  const makeFirestoreSnapshot = (app.firestore() as any).snapshot_.bind(app.firestore());
 
   const resourcePath = proto.context.resource.name;
   const params = extractParamsFromPath(trigger.definition.eventTrigger.resource, resourcePath);
 
   /*
-  We use an internal Firestore method snapshot_ to generate Snapshots which we pass to firebase-function's
+  We use an internal Firestore method makeFirestoreSnapshot to generate Snapshots which we pass to firebase-function's
   Change object. If we have a value for new / old snap, then we create a valid snapshot, otherwise we
-  invoke snapshot_ with a different signature describe here...
+  invoke makeFirestoreSnapshot with a different signature describe here...
   https://github.com/googleapis/nodejs-firestore/blob/114de25d1af3fc7441da3242f6bc8cc8354ffa09/dev/test/index.ts#L574
   To create a snapshot where .exists() fails.
    */
 
   let newSnap;
   if (proto.data.value) {
```

Comment:
Probably out of scope for this PR but I think this block, stubbing the snapshot making, is what was making it hard for me to do callable functions with Firestore.

I found myself having to stuff a `name` into `data.value` and `data.oldValue` and I didn't see any code like that in the old functions emu.